### PR TITLE
When no cycle are used in the scenario configs ...

### LINKED ISSUE #1343: Scenario Selector - Default filters should be consistent with config
### Description

The goal of this issue is to improve the UX of Scenario Selector filters. Some filters are for example not useful when not using cycles:

- primary
- cycle label
- cycle start
- cycle end

These filters should not be shown as they were not used in this case.

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
do not show Cycle* and primary as column for filtering 
resolves #1343

### CODE REVIEW

File: taipy/gui_core/_adapters.py [hunk 1]
```diff
@@ -257,17 +257,17 @@ def _get_datanode_property(attr: str):
...
-    _SC_TYPES = {
-        "Config id": "string",
-        "Label": "string",
-        "Creation date": "date",
-        "Cycle label": "string",
-        "Cycle start": "date",
-        "Cycle end": "date",
-        "Primary": "boolean",
-        "Tags": "string",
+    _SC_TYPES: t.Dict[str, t.Dict[str, t.Union[str, bool]]] = {
```
Comment (FabienLelaquais) [inline taipy/gui_core/_adapters.py]: Now that I think again about this, I think it makes far more sense to have a single dict:
```
_SC_PROPS: t.Dict[str, t.Dict[str, t.Union[str, bool]]] = {
    "Config id": {"attr": "config_id", "type": "string"},
    "Label": {"attr": "name", "type": "string"},
    "Creation date": {"attr": "creation_date", "type": "date"},
    ...
    "Cycle label": {"attr": "cycle.name", "type": "string", "for_cycle": True},
    ...
}
```

No need for _TYPES and _LABELS really, and less error prone.
Note that I renamed "cycle" to "for_cycle". Less ambiguous.

File: taipy/gui_core/_adapters.py [hunk 2]
```diff
@@ -289,7 +289,7 @@ def get_hash():
     def get_type(attr: str):
         if prop := _get_datanode_property(attr):
             return _GuiCoreScenarioProperties.__DN_TYPES.get(prop, "any")
-        return _GuiCoreScenarioProperties._SC_TYPES.get(attr, "any")
+        return _GuiCoreScenarioProperties._SC_TYPES.get(attr, {"type": "any"}).get("type", "any")
```
Comment (FabienLelaquais) [inline taipy/gui_core/_adapters.py]: That reads really strange: if 'attr' is unknown, you carry on anyway?
With my renaming I would propose `_SC_PROPS.get(attr).get("type", "any")`

File: taipy/gui_core/_adapters.py [hunk 3]
```diff
@@ -342,6 +342,9 @@ def get(self):
 
 class _GuiCoreScenarioFilter(_GuiCoreScenarioProperties):
     DEFAULT = list(_GuiCoreScenarioProperties._SC_TYPES.keys())
+    DEFAULT_NO_CYCLE = [
+        p[0] for p in filter(lambda prop: not prop[1].get("cycle", False), _GuiCoreScenarioProperties._SC_TYPES.items())
```
Comment (FabienLelaquais) [inline taipy/gui_core/_adapters.py]:
```suggestion
        p[0] for p in filter(lambda prop: not prop[1].get("for_cycle", False), _GuiCoreScenarioProperties._SC_TYPES.items())
```

**Review by FabienLelaquais — DISMISSED** [review]
Let me know if you feel that my recommendation makes sense...

File: taipy/gui_core/_adapters.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply taipy/gui_core/_adapters.py]: Support for custom properties!

**Review by dinhlongviolin1 — APPROVED** [review]
LGTM

**Review by FabienLelaquais — APPROVED** [review]
👍

File: taipy/gui_core/_adapters.py (hunk 2 above)
Comment (FabienLelaquais) [reply taipy/gui_core/_adapters.py]: Exxxxxxxxxxcellent point

**Pull request merged**